From a80fad2140dcf9bf7b789beef4389ca190055ac3 Mon Sep 17 00:00:00 2001
From: Michael Bestas <mikeioannina@cyanogenmod.org>
Date: Thu, 4 Aug 2016 21:28:53 +0300
Subject: [PATCH 6/8] Snap: Make developer menu more accessible

 * Not all devices have red eye reduction, which made it
   impossible to enable advanced options. Move the toggle
   to max brightness option which is present in all devices.
 * Allow enabling advanced options through camcorder settings.
 * Move hardcoded strings to cm_strings for translations.
 * Close menus when developer mode is toggled so we can see
   the extra options when the menu is reopened.
 * Decrease taps to 7, matching Settings tap-to-enable.
 * Properly reset tap counter.

Change-Id: Iac39b9309388b92bf75a49a6091b483b13bd9154
---
 res/values/cm_strings.xml             |  4 +++
 src/com/android/camera/PhotoMenu.java | 50 +++++++++++++++++++----------------
 src/com/android/camera/VideoMenu.java | 27 +++++++++++++++++++
 3 files changed, 58 insertions(+), 23 deletions(-)

diff --git a/res/values/cm_strings.xml b/res/values/cm_strings.xml
index f152ea4..f5c370c 100644
--- a/res/values/cm_strings.xml
+++ b/res/values/cm_strings.xml
@@ -144,4 +144,8 @@
     <string name="pref_camera_max_brightness_title">Bright screen</string>
     <string name="pref_camera_max_brightness_default" translatable="false">off</string>
 
+    <!-- Advanced options -->
+    <string name="developer_menu_enabled">You have enabled advanced camera options</string>
+    <string name="developer_menu_disabled">You have disabled advanced camera options</string>
+
 </resources>
diff --git a/src/com/android/camera/PhotoMenu.java b/src/com/android/camera/PhotoMenu.java
index ef6e256..6637e05 100644
--- a/src/com/android/camera/PhotoMenu.java
+++ b/src/com/android/camera/PhotoMenu.java
@@ -96,7 +96,7 @@ public class PhotoMenu extends MenuController
     private static final int MODE_SCENE = 0;
     private static final int MODE_FILTER = 1;
     private static final int MODE_MAKEUP = 2;
-    private static final int DEVELOPER_MENU_TOUCH_COUNT = 10;
+    private static final int DEVELOPER_MENU_TOUCH_COUNT = 7;
     private int mSceneStatus;
     private View mHdrSwitcher;
     private View mTsMakeupSwitcher;
@@ -115,7 +115,7 @@ public class PhotoMenu extends MenuController
     private String mPrevSavedCDS;
     private boolean mIsTNREnabled = false;
     private boolean mIsCDSUpdated = false;
-    private int privateCounter = 0;
+    private int mPrivateCounter = 0;
     private static final int ANIMATION_DURATION = 300;
     private static final int CLICK_THRESHOLD = 200;
     private int previewMenuSize;
@@ -1375,27 +1375,6 @@ public class PhotoMenu extends MenuController
     }
 
     public void onPreferenceClicked(ListPreference pref, int y) {
-        // Developer menu
-        if (pref.getKey().equals(CameraSettings.KEY_REDEYE_REDUCTION)) {
-            privateCounter++;
-            if (privateCounter >= DEVELOPER_MENU_TOUCH_COUNT) {
-                SharedPreferences prefs = PreferenceManager
-                        .getDefaultSharedPreferences(mActivity);
-                if (!mActivity.isDeveloperMenuEnabled()) {
-                    mActivity.enableDeveloperMenu();
-                    prefs.edit().putBoolean(CameraSettings.KEY_DEVELOPER_MENU, true).apply();
-                    RotateTextToast.makeText(mActivity,
-                            "Camera developer option is enabled now", Toast.LENGTH_SHORT).show();
-                } else {
-                    mActivity.disableDeveloperMenu();
-                    prefs.edit().putBoolean(CameraSettings.KEY_DEVELOPER_MENU, false).apply();
-                    RotateTextToast.makeText(mActivity,
-                            "Camera developer option is disabled now", Toast.LENGTH_SHORT).show();
-                }
-            }
-        } else {
-            privateCounter = 0;
-        }
         LayoutInflater inflater = (LayoutInflater) mActivity.getSystemService(
                 Context.LAYOUT_INFLATER_SERVICE);
         ListSubMenu basic = (ListSubMenu) inflater.inflate(
@@ -1411,6 +1390,31 @@ public class PhotoMenu extends MenuController
             mUI.showPopup(mListSubMenu, 2, true);
         }
         mPopupStatus = POPUP_SECOND_LEVEL;
+
+        // Developer menu
+        if (pref.getKey().equals(CameraSettings.KEY_MAX_BRIGHTNESS)) {
+            mPrivateCounter++;
+            if (mPrivateCounter >= DEVELOPER_MENU_TOUCH_COUNT) {
+                SharedPreferences prefs = PreferenceManager
+                        .getDefaultSharedPreferences(mActivity);
+                if (!mActivity.isDeveloperMenuEnabled()) {
+                    mActivity.enableDeveloperMenu();
+                    prefs.edit().putBoolean(CameraSettings.KEY_DEVELOPER_MENU, true).apply();
+                    closeAllView();
+                    RotateTextToast.makeText(mActivity,
+                            R.string.developer_menu_enabled, Toast.LENGTH_SHORT).show();
+                } else {
+                    mActivity.disableDeveloperMenu();
+                    prefs.edit().putBoolean(CameraSettings.KEY_DEVELOPER_MENU, false).apply();
+                    closeAllView();
+                    RotateTextToast.makeText(mActivity,
+                            R.string.developer_menu_disabled, Toast.LENGTH_SHORT).show();
+                }
+                mPrivateCounter = 0;
+            }
+        } else {
+            mPrivateCounter = 0;
+        }
     }
 
     public void onListMenuTouched() {
diff --git a/src/com/android/camera/VideoMenu.java b/src/com/android/camera/VideoMenu.java
index 3543c91..e486017 100644
--- a/src/com/android/camera/VideoMenu.java
+++ b/src/com/android/camera/VideoMenu.java
@@ -78,6 +78,7 @@ public class VideoMenu extends MenuController
     private static final int PREVIEW_MENU_IN_ANIMATION = 1;
     private static final int PREVIEW_MENU_ON = 2;
     private static final int MODE_FILTER = 1;
+    private static final int DEVELOPER_MENU_TOUCH_COUNT = 7;
     private int mSceneStatus;
     private View mFrontBackSwitcher;
     private View mFilterModeSwitcher;
@@ -87,6 +88,7 @@ public class VideoMenu extends MenuController
     private String mPrevSavedVideoCDS;
     private boolean mIsVideoTNREnabled = false;
     private boolean mIsVideoCDSUpdated = false;
+    private int mPrivateCounter = 0;
     private static final int ANIMATION_DURATION = 300;
     private static final int CLICK_THRESHOLD = 200;
     private int previewMenuSize;
@@ -901,6 +903,31 @@ public class VideoMenu extends MenuController
             mUI.showPopup(mListSubMenu, 2, true);
         }
         mPopupStatus = POPUP_SECOND_LEVEL;
+
+        // Developer menu
+        if (pref.getKey().equals(CameraSettings.KEY_MAX_BRIGHTNESS)) {
+            mPrivateCounter++;
+            if (mPrivateCounter >= DEVELOPER_MENU_TOUCH_COUNT) {
+                SharedPreferences prefs = PreferenceManager
+                        .getDefaultSharedPreferences(mActivity);
+                if (!mActivity.isDeveloperMenuEnabled()) {
+                    mActivity.enableDeveloperMenu();
+                    prefs.edit().putBoolean(CameraSettings.KEY_DEVELOPER_MENU, true).apply();
+                    closeAllView();
+                    RotateTextToast.makeText(mActivity,
+                            R.string.developer_menu_enabled, Toast.LENGTH_SHORT).show();
+                } else {
+                    mActivity.disableDeveloperMenu();
+                    prefs.edit().putBoolean(CameraSettings.KEY_DEVELOPER_MENU, false).apply();
+                    closeAllView();
+                    RotateTextToast.makeText(mActivity,
+                            R.string.developer_menu_disabled, Toast.LENGTH_SHORT).show();
+                }
+                mPrivateCounter = 0;
+            }
+        } else {
+            mPrivateCounter = 0;
+        }
     }
 
     public void onListMenuTouched() {
-- 
2.7.4

